### PR TITLE
MODOAIPMH-379: Gracefully handle JsonParse exception in case of 403 response from inventory-storage

### PR DESCRIPTION
**Pull request by IlliaDaliek**
Jira: https://issues.folio.org/browse/MODOAIPMH-379

### PURPOSE
Gracefully handle JsonParse exception in case of 403 response from inventory-storage and put the correct message with missing permissions to the response.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -17,9 +17,6 @@
             "source-storage.records.get",
             "source-storage.sourceRecords.get",
             "configuration.entries.collection.get",
-            "inventory-storage.oai-pmh-view.instances.collection.get",
-            "inventory-storage.oai-pmh-view.updatedinstanceids.collection.get",
-            "inventory-storage.oai-pmh-view.enrichedinstances.collection.post",
```
Comment (IlliaDaliek) [inline descriptors/ModuleDescriptor-template.json:22]: oai-pmh view is not used anymore

File: src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java [hunk 2]
```diff
@@ -94,55 +94,49 @@
...
-  private  MetricsCollectingService metricsCollectingService = MetricsCollectingService.getInstance();
-  private InstancesService instancesService;
+  public static final MarcWithHoldingsRequestHelper INSTANCE = new MarcWithHoldingsRequestHelper();
+
+  private final Vertx vertx;
   private final WorkerExecutor saveInstancesExecutor;
   private final Context downloadContext;
 
+  private MetricsCollectingService metricsCollectingService = MetricsCollectingService.getInstance();
+  private InstancesService instancesService;
```
Comment (IlliaDaliek) [inline src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java:138]: Just corrected the order of variables declaration to have better readability and view.

File: src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java [hunk 3]
```diff
@@ -297,21 +295,37 @@ private void setupBatchHttpStream(Promise<?> promise, HttpRequestImpl<Buffer> in
       }
       downloadInstancesPromise.complete();
     });
-    jsonParser.exceptionHandler(throwable -> {
-      logger.error("Error has been occurred at JsonParser while reading data from response. Message: {}", throwable.getMessage(),
+    jsonParser.exceptionHandler(throwable -> responseChecked.future().onSuccess(invalidResponseReceivedAndProcessed -> {
+        if (invalidResponseReceivedAndProcessed) {
```
Comment (IlliaDaliek) [inline src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java:299]: I don't think that it is a code smell. I can add `invalidResponseReceivedAndProcessed.booleanValue()` if you think that this matters.

File: src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java [hunk 4]
```diff
@@ -473,29 +488,46 @@ private void getNextBatch(String requestId, Request request, boolean firstBatch,
         logger.info("Instance with instanceId {} wasn't in the request.", instanceId);
       }
     });
-    jsonParser.exceptionHandler(throwable -> {
-      logger.error("Error has been occurred at JsonParser while reading data from response. Message:{}", throwable.getMessage(),
+    jsonParser.exceptionHandler(throwable -> responseChecked.future().onSuccess(invalidResponseReceivedAndProcessed -> {
+        if (invalidResponseReceivedAndProcessed) {
```
Comment (IlliaDaliek) [inline src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java:492]: I don't think that it is a code smell. I can add `invalidResponseReceivedAndProcessed.booleanValue()` if you think that this matters.

**Review by alekGbuz — APPROVED** [review]

**Review by siarhei-charniak — APPROVED** [review]

**Pull request merged**